publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -142,7 +142,27 @@ private static Resource convertToResource(Publication publication) {
                    .withDuplicateOf(publication.getDuplicateOf())
                    .withCuratingInstitutions(publication.getCuratingInstitutions())
                    .withImportDetails(publication.getImportDetails())
-                   .build();
+                   .build().validate();
+    }
+
+    private Resource validate() {
+        var cristinIdentifiers = getCristinIdentifiers();
+        if (hasMultipleEntries(cristinIdentifiers)) {
+            throw new MultipleCristinIdentifiersException("Multiple cristin identifiers are not allowed!");
+        } else {
+            return this;
+        }
+    }
+
```

Comment:
Adding following validation to `Resource.fromPublication()` method so we always validate dao to persist.